bertie_bogo_shim/src/main.rs

```diff
@@ -0,0 +1,214 @@
+// # Bertie BoGo Shim
+//
+// # Credit
+//
+// The code is based on rustls' bogo shim ...
+//
+//     https://github.com/rustls/rustls/blob/main/rustls/examples/internal/bogo_shim.rs
+//
+
+extern crate core;
+
+use std::{env, net::TcpStream, process};
+use tracing::Level;
+
+use simple_https_client::tls13client;
+
+static BOGO_NACK: i32 = 89;
+
+#[derive(Debug)]
+struct Options {
+    port: u16,
+    role: Role,
+    key_file: String,
+    cert_file: String,
+    hostname: String,
+    expect_fallback_scsv: bool,
+    expect_extended_master_secret: bool,
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Options {
+            port: 0,
+            role: Role::Client,
+            key_file: "".into(),
+            cert_file: "".into(),
+            hostname: "example.com".into(),
+            expect_fallback_scsv: false,
+            expect_extended_master_secret: false,
+        }
+    }
+}
+
+#[derive(Debug)]
+enum Role {
+    Client,
+    Server,
+}
+
+const UNHANDLED_ARGUMENTS: &[&str] = &[
+    "-advertise-alpn",
+    "-async",
+    "-check-close-notify",
+    "-cipher",
+    "-curves",
+    "-ech-config-list",
+    "-enable-early-data",
+    "-enable-ech-grease",
+    "-enable-grease",
+    "-enable-ocsp-stapling",
+    "-enforce-rsa-key-usage",
+    "-expect-certificate-types",
+    "-expect-cipher-aes",
+    "-expect-client-ca-list",
+    "-expect-curve-id",
+    "-expect-no-session",
+    "-expect-peer-cert-file",
+    "-expect-peer-signature-algorithm",
+    "-expect-peer-verify-pref",
+    "-expect-secure-renegotiation",
+    "-expect-ticket-supports-early-data",
+    "-export-keying-material",
+    "-export-traffic-secrets",
+    "-fail-cert-callback",
+    "-fail-early-callback",
+    "-fips-202205",
+    "-forbid-renegotiation-after-handshake",
+    "-implicit-handshake",
+    "-install-cert-compression-algs",
+    "-key-update",
+    "-ocsp-response",
+    "-on-resume-expect-accept-early-data",
+    "-on-resume-expect-reject-early-data",
+    "-peek-then-read",
+    "-psk",
+    "-read-with-unfinished-write",
+    "-renegotiate-explicit",
+    "-renegotiate-freely",
+    "-renegotiate-ignore",
+    "-renegotiate-once",
+    "-require-any-client-certificate",
+    "-resume-count",
+    "-send-channel-id",
+    "-shim-shuts-down",
+    "-shim-writes-first",
+    "-signing-prefs",
+    "-srtp-profiles",
+    "-tls-unique",
+    "-use-old-client-cert-callback",
+    "-verify-fail",
+    "-verify-peer",
+    "-verify-prefs",
+    "-write-settings",
+];
+
+fn main() {
```

Comment:
Please document all this code.

Reply:
I'm not sure if it is the right time to add the documentation. This will very likely change anyway. I think we should just keep it for now. I'll revisit as soon as I start working on it again.

Reply:
Please stop force pushing. This makes reviewing impossible.
Code will always change. But code also must be documented. You're not the only one working on this and everyone has to know what it's about.

Reply:
See https://github.com/cryspen/bertie/pull/41/commits/d9ee3d0a6164a59b572c6eb249fa38ab51201faa